src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2477,6 +2535,16 @@ private void readObject(ObjectInputStream s)
             }
         }
 
+        @Override
+        public int hashCode() {
+            return (style != null) ? style.hashCode() : 0;
+        }
+
+        @Override
+        public boolean equals(Object val) {
+            return val instanceof CSS.BorderStyle border && style.equals(border.style);
```

Comment:
The `style` field is of type `CSS.Value` which doesn't implement `equals` (yet). It shouldn't be a problem, however, you may use `==` explicitly.